pkg/crypto.go

```diff
@@ -301,20 +310,39 @@ func (client *Crypto) SignFor(data []byte, legalEntity types.LegalEntity) ([]byt
 	return signature, err
 }
 
+// KeyExistsFor checks storage for an entry for the given legal entity and returns true if it exists
+func (client *Crypto) KeyExistsFor(legalEntity types.LegalEntity) bool {
+	if _, err := client.Storage.GetPrivateKey(legalEntity); err != nil {
+		return false
+	}
+	return true
+}
+
 // VerifyWith verfifies a signature of some data with a given PublicKeyInPEM. It uses the SHA256 hashing function.
-func (client *Crypto) VerifyWith(data []byte, sig []byte, pemKey string) (bool, error) {
-	key, err := PemToPublicKey([]byte(pemKey))
+func (client *Crypto) VerifyWith(data []byte, sig []byte, key jwk.Key) (bool, error) {
+	hashedData := sha256.Sum256(data)
```

Comment:
should the hashing algorithm be so hardcoded?

Reply:
I looked at what to do in case of an EC based signature. It requires different arguments. So I think a switch to JWS might be required in the future somewhere.....

Reply:
ok